recipes/mosquitto/2.x/conanfile.py

```diff
@@ -76,61 +70,38 @@ def requirements(self):
-        self._cmake.definitions["WITH_WEBSOCKETS"] = self.options.get_safe("websockets", False)
-        self._cmake.definitions["STATIC_WEBSOCKETS"] = self.options.get_safe("websockets", False) and not self.options["libwebsockets"].shared
-        self._cmake.definitions["DOCUMENTATION"] = False
-        self._cmake.definitions["CMAKE_INSTALL_SYSCONFDIR"] = os.path.join(self.package_folder, "res").replace("\\", "/")
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    def _patch_sources(self):
-        replace_in_file(self, os.path.join(self._source_subfolder, "client", "CMakeLists.txt"), "static)", "static ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "client", "CMakeLists.txt"), "quitto)", "quitto ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "static)", "static ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "quitto)", "quitto ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_passwd", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "MOSQ_LIBS", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "include", "mosquitto.h"), "__declspec(dllimport)", "")
-        replace_in_file(self, os.path.join(self._source_subfolder, "lib", "cpp", "mosquittopp.h"), "__declspec(dllimport)", "")
-        # dynlibs for apple mobile want code signatures and that will not work here
-        # this would actually be the right patch for static builds also, but this would have other side effects, so
-        if(self.settings.os in ["iOS", "watchOS", "tvOS"]):
-            replace_in_file(self, os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"), "SHARED", "")
-            replace_in_file(self, os.path.join(self._source_subfolder, "lib", "cpp", "CMakeLists.txt"), "SHARED", "")
+            tc.variables["WITH_CJSON"] = self.options.get_safe("cjson")
+        tc.variables["WITH_BROKER"] = self.options.broker
+        tc.variables["WITH_APPS"] = self.options.apps
+        tc.variables["WITH_PLUGINS"] = False
+        tc.variables["WITH_LIB_CPP"] = self.options.build_cpp
+        tc.variables["WITH_THREADING"] = self.settings.compiler != "msvc" and self.options.threading
```

Comment:
For 1.x and 2.0 support :)

```suggestion
        tc.variables["WITH_THREADING"] = is_msvc(self) and self.options.threading
```